2-extra/3-groceries.js

```diff
@@ -28,13 +28,27 @@ Exercise 1:
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
 
+for (let key in weeklyMealPlan) {
+  for (let i = 0; i < weeklyMealPlan[key].length; i++) {
+    if (weeklyGroceriesToBuy.indexOf(weeklyMealPlan[key][i]) === -1) {
+      weeklyGroceriesToBuy.push(weeklyMealPlan[key][i]);
+    }
```

Comment:
This works fine, but usually we would use sets if we don't want repeating items.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set